sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -385,6 +380,8 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
         }
         EliminateSubqueryAliases(catalog.lookupRelation(tableIdentWithDB)) match {
           // Only do the check if the table is a data source table (the relation is a BaseRelation).
+          // TODO(cloud-fan): also check hive table relation here when we support overwrite mode
+          // for creating hive tables.
```

Comment:
We should not block it. This generates `InsertIntoTable`, and it supports hive table. What we should block is `saveAsTable` with `Overwrite` mode, which generates `CreateTable`.

`insert overwrite` is different from `create table with overwrite mode`